src/main/java/com/st/lms/service/BorrowerService.java

```diff
@@ -0,0 +1,172 @@
+package com.st.lms.service;
+
+import java.sql.Date;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.st.lms.dao.AuthorDao;
+import com.st.lms.dao.BookCopiesDao;
+import com.st.lms.dao.BookDao;
+import com.st.lms.dao.BookLoansDao;
+import com.st.lms.dao.BorrowerDao;
+import com.st.lms.dao.LibBranchDao;
+import com.st.lms.dto.BkCopiesDTO;
+import com.st.lms.dto.BkLoansBkAuthDTO;
+import com.st.lms.dto.BkLoansBranchDTO;
+import com.st.lms.models.Author;
+import com.st.lms.models.Book;
+import com.st.lms.models.BookCopies;
+import com.st.lms.models.BookCopiesPrimaryKey;
+import com.st.lms.models.BookLoans;
+import com.st.lms.models.BookLoansPrimaryKey;
+import com.st.lms.models.LibraryBranch;
+import com.st.lms.utils.DateCalculations;
+
+@Service
+public class BorrowerService {
+	
+	@Autowired
+	private BookDao bookDao;
+	@Autowired
+	private BookLoansDao bookLoansDao;
+	@Autowired
+	private BookCopiesDao bookCopiesDao;
+	@Autowired
+	private BorrowerDao borrowerDao;
+	@Autowired
+	private LibBranchDao libBranchDao;
+	@Autowired
+	private AuthorDao authorDao;
+	
+	public boolean cardNoExists(int cardNo) {
+		return borrowerDao.findById(cardNo).isPresent();
+	}
+	
+	public List<LibraryBranch> getAllBranches() {
+		List<LibraryBranch> libBranches = null;
+		libBranches = libBranchDao.findAll();
+		return libBranches;
+	}
+	
+	public boolean branchExists(int branchId) {
+		return libBranchDao.findById(branchId).isPresent();
+	}
+	
+	public boolean loanExists(int cardNo, int branchId, int bookId) {
+		return bookLoansDao.findById(new BookLoansPrimaryKey(bookId, branchId, cardNo)).isPresent();
+	}
+	
+	//returns all bookCopies>=1 specific to a branch with book and author
+	public List<BkCopiesDTO> getBkCopiesGreater1BookAndTitle(int branchId) {
+		List<BkCopiesDTO> list = null;

```

Comment:
Move the declaration down to where this is initialized rather than declaring it `= null`.